.github/workflows/release-assignment.yml

```diff
@@ -74,17 +77,56 @@ jobs:
               core.setFailed(`Failed to fetch calendar events: ${error.message}`);
             }
 
+      - name: Get all events for version
+        id: get-all-events-for-version
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const ical = require('node-ical');
+            const version = '${{ steps.check-code-freeze-8-weeks.outputs.version }}';
+
+            const releaseEvents = {
+              'beta-1': `WooCommerce ${version} Beta 1`,
+              'beta-2': `WooCommerce ${version} Beta 2`,
+              'rc-1': `WooCommerce ${version} RC 1`,
+              final: `WooCommerce ${version} Release`,
+            };
+
+            const events = await ical.async.fromURL(
+              `https://calendar.google.com/calendar/ical/${{ secrets.RELEASE_CALENDAR_ID }}/public/basic.ics`
+            );
+
+            const calendarEvents = Object.values(events)
+              .filter(event => event.type === 'VEVENT')
+              .filter(event => event.summary && event.start);
+
+            for (const [eventKey, eventTitle] of Object.entries(releaseEvents)) {
+              const matchingEvent = calendarEvents.find(event => 
+                event.summary.trim() === eventTitle
+              );
+
+              const date = matchingEvent ? new Date(matchingEvent.start).toLocaleDateString('en-US') : null;
+              core.setOutput(`event-${eventKey}-date`, date);
```

Comment:
Would it make sense to convert these dates to YYYY-MM-DD so there's 0 chance of confusion when reading the issue?

<img width="290" height="108" alt="Screenshot 2025-08-08 at 17 03 20" src="https://github.com/user-attachments/assets/864d2b67-31a2-4d24-af1d-f183ba82cdd5" />

Reply:
Update in c65e7d68824e30187656f89507018dd3490f3293